src/main/scala/workflow/Pipeline.scala

```diff
@@ -0,0 +1,242 @@
+package workflow
+
+import org.apache.spark.rdd.RDD
+import pipelines.Logging
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+trait Pipeline[A, B] {
+  private[workflow] val nodes: Seq[Node]
```

Comment:
These member variables need comments -- what is a fitDep, what is a dataDep etc. In fact an over-arching comment like `each pipeline consists of a number of nodes and a special sink node ...`  might be useful